app/Http/Controllers/SettingController.php

```diff
@@ -2,9 +2,86 @@
 
 namespace App\Http\Controllers;
 
+use App\Http\Requests\StoreSettingRequest;
+use App\Models\Setting;
+use App\Models\User;
 use Illuminate\Http\Request;
+use Illuminate\Validation\Rule;
 
 class SettingController extends Controller
 {
-    //
+    public function index()
+    {
+        // get Sam to consolidate this, it could be improved, maybe a different way instead of get()?
+        return view('settings.index',[
+            'settings' => auth()->user()->settings()->get()
+        ]);
+
+    }
+
+    public static function create(User $user, string $key, string $value)
+    {
+        //creates a user setting from the parameters. use in code only. will be redundant probably.
+
+        $user->settings()->create([
+            'user_id' => $user->id,
```

Comment:
I recommend using `$model->getKey()` just in case we ever change the primary key of the model, we can change it in one place

Reply:
But be careful not to do it with relationships because `$model->relation->getKey()` will actually pull the relationship out of the database. When dealing with relationships, it's nicer to do `$model->relation_id`.